src/main/java/com/owncloud/android/ui/dialog/ConflictsResolveDialog.java

```
@@ -40,49 +65,120 @@
  */
 public class ConflictsResolveDialog extends DialogFragment {
 
+    private OCFile existingFile;
+    private File newFile;
+    private OnConflictDecisionMadeListener listener;
+    private User user;
+    private List<ThumbnailsCacheManager.ThumbnailGenerationTask> asyncTasks = new ArrayList<>();
+
     public enum Decision {
         CANCEL,
         KEEP_BOTH,
         KEEP_LOCAL,
         KEEP_SERVER,
     }
 
-    private final OnConflictDecisionMadeListener listener;
-    private final boolean canKeepServer;
-
-    public ConflictsResolveDialog(OnConflictDecisionMadeListener listener, boolean canKeepServer) {
+    public ConflictsResolveDialog(OnConflictDecisionMadeListener listener,
+                                  OCFile file,
+                                  OCUpload conflictUpload, Optional<User> user) {
         this.listener = listener;
-        this.canKeepServer = canKeepServer;
+        this.existingFile = file;
+        this.newFile = new File(conflictUpload.getLocalPath());
+        this.user = user.get();
```

This `Optional<T>.get()` can throw NPE.
If we really, really, really know it won't be null, let's use `getOrElse(IllegalStateException::new)`.

The main reason we introduced `Optional<User>` was to replace nullable `getAccount()` call. The trouble with `getAccount()` was that callers were liberally assuming valid return value. Since the code was then full of `if (account == null) { some ad-hoc remediation }` workarounds, this assumption was not really holding up. My suspicion was that we were experiencing many edge cases of NPEs and were trying to patch them in some simple way.

Hence the `Optional<T>` which provides more explicit and harder to ignore API.

Having said all of the above, I think that the best idea would be to pass `User` instead of `Optional<User>` and just make sure that if the user dependency is not available, we never get here in the first place.

---

Ok, will pass User and fail early, if no user is available.